Classes/ToolRegistrator.lua

```diff
@@ -69,12 +69,18 @@ function cToolRegistrator:BindAbsoluteTools()
 		return true
 	end
 
+	local LastRightClick = -math.huge
 	local function OnPlayerRightClick(a_Player, a_BlockX, a_BlockY, a_BlockZ, a_BlockFace)
 		local Succ, Message = GetPlayerState(a_Player).Selection:SetPos(a_BlockX, a_BlockY, a_BlockZ, a_BlockFace, "Second")
 		if (not Succ) then
 			return false
 		end
 
+		if ((os.clock() - LastRightClick) < 0.005) then
```

Comment:
A comment here explaining why this is necessary would be useful (multiple packets come through in a single tick?)

Reply:
> A comment here explaining why this is necessary would be useful (multiple packets come through in a single tick?)

In some cases, when the second position is indicated, the client sends 2 packets

Reply:
Could you check the `a_BlockFace` parameter? It's possible we can filter out the second packet if `a_BlockFace` is `BLOCK_FACE_NONE`.

Also, wouldn't it make more sense to put the check before setting the selection? Now the selection still gets set twice, but the second time the player isn't notified.

Reply:
> Could you check the `a_BlockFace` parameter? It's possible we can filter out the second packet if `a_BlockFace` is `BLOCK_FACE_NONE`.
> 
> Also, wouldn't it make more sense to put the check before setting the selection? Now the selection still gets set twice, but the second time the player isn't notified.

The problem is that we catch 3 packages, 2 of them are normal, and the third with BLOCK_FACE_NONE. Even with verification, 2 messages are displayed. Example:
1. Normal
2. BLOCK_FACE_NONE
3. Normal

Reply:
I see. I guess the `os.clock()` method is the simplest way to avoid this. It's also somewhat forwards compatible if Mojang decide to change the number of packets again.